stix_shifter/stix_transmission/src/modules/carbonblack/carbonblack_connector.py

```diff
@@ -0,0 +1,62 @@
+from ..base.base_connector import BaseConnector
+from .carbonblack_api_client import APIClient
+import json
+
+
+class Connector(BaseConnector):
+    def __init__(self, connection, configuration, dialect="process"):
+        if dialect not in ["process", "binary"]:
+            raise NotImplementedError("Invalid module dialect")
+
+        self.api_client = APIClient(connection, configuration, dialect=dialect)
+        self.ping_connector = self
+        self.results_connector = self
+        self.query_connector = self
+        self.is_async = False
+
+    def ping(self):
+        try:
+            response = self.api_client.ping_box()
+            response_code = response.code
+            response_json = json.loads(response.read())
+
+            return_obj = dict()
+
+            if len(response_json) > 0 and 200 <= response_code < 300:
+                return_obj['success'] = True
+            else:
+                return_obj['success'] = False
```

Comment:
error answer is supposed to have an 'code' field. Please check other modules(qradar,bigfix and splunk) for error reporting. It may be needed to add error_mapper.py to implement proper error mapping.

Reply:
Ok. I'll add that as well.